### DISCUSSION TITLE
Left over tasks

### DISCUSSION THREAD

**Discussion by lovgrandma** [hibiken/asynq#465] [Q&A]
_How do I clear the left over tasks in queue?_ I have some left over tasks from when I didnt know how to use the job queue. Now Im not sure if my tasks are successfully being deleted from my redis queue after completion.

_Also, how big can your queue be?_ I would ideally like for it to be 1m queued jobs.
I am expecting to have to intentionally slow down the execution of jobs as it involves Mass SMS blasting to subscribed users and carriers are definitely going to flag my customers if they send to many texts too quickly and I am thinking I might have to buffer each message send by exponentially growing time.

**Comment by hibiken** [comment]
@lovgrandma Thank you for opening a discussion!

> How do I clear the left over tasks in queue?

You can use either Web UI or CLI to do a clean up. Refer to the tools README for more info.

 > Also, how big can your queue be?

Currently, we don't have a feature to apply back pressure when enqueueing tasks, but I can imagine that you could use `Inspector.GetQueueInfo` to get queue size and apply some kind of back pressure when queue size reaches certain threshold.

LMK if you have more questions! Happy to help
